server/src/main/java/org/opensearch/action/search/SearchUtils.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.action.search;
+
+import org.opensearch.action.StepListener;
+import org.opensearch.cluster.ClusterState;
+import org.opensearch.cluster.node.DiscoveryNode;
+import org.opensearch.transport.RemoteClusterService;
+
+import java.util.Set;
+import java.util.function.BiFunction;
+
+/**
+ * Helper class for common search functions
+ */
+public class SearchUtils {
+
+    public SearchUtils() {}
+
+    /**
+     * Get connection lookup listener for list of clusters passed
+     */
+    public static StepListener<BiFunction<String, String, DiscoveryNode>> getConnectionLookupListener(
```

Comment:
This doesn't look generic enough, espl that it is in a utility class. Can we have it return a plain `ActionListener` instead